contracts/permissionless/Orders.sol

```diff
@@ -138,6 +126,46 @@ contract Orders is Withdrawable, Utils2, OrdersInterface {
         return false;
     }
 
+    function updateAmounts(
+        uint32 orderId,
+        uint128 srcAmount,
+        uint128 dstAmount
+    )
+        public
+        onlyAdmin
+        returns (bool)
+    {
+        require(orderId != HEAD_ID && orderId != TAIL_ID);
```

Comment:
code for this function is a copy of the code in isRightPosition

if isRightPosition will have another parameter (nextId) u will be able to use it for this.

Reply:
updateAmounts merged into updateWithPositionHint, using a single validation function as we discussed f2f.